common/static/common/js/discussion/views/discussion_topic_menu_view.js

```diff
@@ -45,7 +45,8 @@
             renderCategoryMap: function(map) {
                 var categoryTemplate = edx.HtmlUtils.template($('#new-post-menu-category-template').html()),
                     entryTemplate = edx.HtmlUtils.template($('#new-post-menu-entry-template').html()),
-                    mappedCategorySnippets = _.map(map.children, function(name) {
+                    mappedCategorySnippets = _.map(map.children, function(nameType) {
```

Comment:
I had a lot of different comments.  My recommendations would be:
1. Rename `nameType` to `child` in all code similar to this code (this seems to be repeated).
2. Add the `child[0]` and `child[1]` comment I detailed above.
3. Add a check for the `type` after `_.has(map.entries, name)` as I documented above.
4. Forget about changing the data structure.  I don't think it is worth adding a TODO that will always be ignored.  If you don't do it now, it will not be done.

Hopefully this is clear.